test/module/irohad/consensus/yac/network_test.cpp

```diff
@@ -25,7 +30,24 @@ namespace iroha {
           notifications = std::make_shared<MockYacNetworkNotifications>();
           async_call = std::make_shared<
               network::AsyncGrpcClient<google::protobuf::Empty>>();
-          network = std::make_shared<NetworkImpl>(async_call);
+          // Custom deleter is necessary since it is necessary to construct
+          // unique_ptr in client_creator in there is not possible to make empty
+          // destructor there
+          stub =
+              std::unique_ptr<iroha::consensus::yac::proto::MockYacStub,
+                              void (*)(
+                                  iroha::consensus::yac::proto::MockYacStub *)>(
+                  new iroha::consensus::yac::proto::MockYacStub(),
+                  [](iroha::consensus::yac::proto::MockYacStub *ptr) {});
```

Comment:
This code can be refactored without an empty deleter. 
https://github.com/hyperledger/iroha/blob/324608ec2cceac4733b8259f18accae97bd07e52/test/module/irohad/ordering/on_demand_connection_manager_test.cpp#L27-L29
Similar to these lines, you make a raw pointer `iroha::consensus::yac::proto::MockYacStub* stub` fixture field instead of a `shared_ptr`. There is no need to create one stub in `SetUp`. The code above can be put in `client_creator`.